misc/python/materialize/cli/mzcompose.py

```diff
@@ -177,142 +189,251 @@ def gen_shortcuts(repo: mzbuild.Repository) -> int:
+class DockerComposeCommand(Command):
+    add_help = False
+
+    def __init__(self, name: str, help: str, acquire_deps: bool = False):
+        self.name = name
+        self.help = help
+        self.acquire_deps = acquire_deps
+
+    def configure(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument("-h", "--help", action="store_true")
+
+    def run(self, args: argparse.Namespace) -> None:
+        if args.help:
+            output = spawn.capture(
+                ["docker-compose", self.name, "--help"], stderr_too=True, unicode=True
+            )
+            output = output.replace("docker-compose", "./mzcompose")
+            print(output, file=sys.stderr)
+            return
+
+        # Make sure Docker Compose is new enough.
+        output = (
+            spawn.capture(["docker-compose", "version", "--short"], unicode=True)
+            .strip()
+            .strip("v")
+        )
+        version = tuple(int(i) for i in output.split("."))
+        if version < MIN_COMPOSE_VERSION:
+            msg = f"Unsupported docker-compose version: {version}, min required: {MIN_COMPOSE_VERSION}"
+            raise errors.MzConfigurationError(msg)
+
+        announce("Collecting mzbuild dependencies")
+        composition = load_composition(args)
+        deps = composition.repo.resolve_dependencies(composition.images)
+        for d in deps:
+            say(d.spec())
+
+        if self.acquire_deps:
+            if args.coverage:
+                # If the user has requested coverage information, create the
+                # coverage directory as the current user, so Docker doesn't create
+                # it as root.
+                (composition.path / "coverage").mkdir(exist_ok=True)
+            deps.acquire()
+
+        self.handle_composition(args, composition)
+
+    def handle_composition(
+        self, args: argparse.Namespace, composition: mzcompose.Composition
+    ) -> None:
+        announce("Delegating to Docker Compose")
+        proc = composition.run(
+            [*args.unknown_args, self.name, *args.unknown_subargs],
+            check=False,
         )
-        self.add_argument("--mz-coverage", action="store_true", default=None)
-        self.add_argument("-f", "--file")
-        self.add_argument("--project-directory")
-        self.add_argument("command", nargs="?")
-        self.add_argument("first_command_arg", nargs="?")
-        self.add_argument("remainder", nargs="*")
-        self.add_argument("--help", action="store_true", default=False)
+        sys.exit(proc.returncode)
+
+
+class RunCommand(DockerComposeCommand):
+    def __init__(self) -> None:
+        super().__init__("run", "run a one-off command")
+
+    def configure(self, parser: argparse.ArgumentParser) -> None:
+        super().configure(parser)
+        parser.add_argument("workflow", nargs="?")
+
+    def run(self, args: argparse.Namespace) -> None:
+        super().run(args)
+        if self.help:
```

Comment:
if you `add_help = True` you could add this help during the `add_parser` call

Reply:
So `add_help = True` adds an argparse managed help message, but we want to roll our own `--help` flag here (via `DockerComposeCommand`) that forwards the help through to Docker Compose.

Reply:
Oh I see, you are also getting the forwarded docker-compose help